deps/wazuh_testing/wazuh_testing/tools/performance/statistic.py

```diff
@@ -101,53 +124,478 @@ def _parse_logcollector_state_file(self, data):
                 file_data['bytes'] = file['bytes']
                 file_data['target'] = target['name']
                 file_data['target_drops'] = target['drops']
-                self._write_csv(file_data, join(self.dst_dir, f'{csv_name}.csv'))
+                self._write_csv(file_data, 'logcollector', join(self.dst_dir, f'{csv_name}.csv'))
+
 
     def _parse_state_file(self):
         """Read the data from the statistics file generated by Wazuh."""
         try:
             logging.info("Getting statistics data from {}".format(self.statistics_file))
-            data = {'Timestamp': datetime.now().strftime('%Y/%m/%d %H:%M:%S')}
+            data = {}
             if not self.parse_json:
-                self._parse_classic_state_file(data)
+                self._parse_classic_state_file(data, self.target)
             else:
                 self._parse_logcollector_state_file(data)
         except Exception as e:
             logger.error(f'Exception with {self.statistics_file} | {str(e)}')
 
-    @staticmethod
-    def _write_csv(data, csv_file):
+
+    def _parse_api_data(self):
+        """Read the data from the statistics file generated by Wazuh API."""
+
+        API_URL = f"https://{self.ip}:{self.port}"
+        DAEMONS_ENDPOINT= f"/manager/daemons/stats?daemons_list={self.daemon}"
+        TOKEN_ENDPOINT="/security/user/authenticate"
+
+        logging.info("Getting statistics data from API for {}".format(self.target))
+
+        max_retries = 3
+        token_response = None
+        daemon_response = None
+        # Try to get the response token three times
+        for _ in range(max_retries):
+            try:
+                token_response = requests.get(API_URL + TOKEN_ENDPOINT, verify=False,
+                                              auth=requests.auth.HTTPBasicAuth("wazuh", "wazuh"))
+                if token_response.status_code != 200:
+                    break
+            except requests.exceptions.RequestException as e:
+                logging.error(f"Error getting token from API: {str(e)}")
+        else:
+            logging.error("Retrying get API data, status code {}".format(token_response.status_code))
+
+        for _ in range(max_retries):
+            try:
+                daemons_response = requests.get(API_URL + DAEMONS_ENDPOINT, verify=False,
+                                                headers={'Authorization': 'Bearer ' + token_response.json()['data']['token']})
+                if daemons_response.status_code != 200:
+                    break
+            except requests.exceptions.RequestException as e:
```

Comment:
Is imperative the use by default of the `wait_for_complete` parameter. Check API reference https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.manager_controller.get_daemon_stats

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/4970/commits/a1d0448b75f63ebcbe1d81489fae944108b17967